### ISSUE TITLE
float64 warning

### ISSUE THREAD

**Issue by apulver** [Theano/Theano#5764]
Warns about creating a float64 when lr1 is used as the learning rate, even though lr1's value was given as a float32. 
```
import theano
import theano.tensor as T
import numpy as np
from collections import OrderedDict


def test():
    X = T.matrix()
    x = np.random.randn(10,32).astype(np.float32)
    m1 = theano.shared(np.random.randn(32,8).astype(np.float32))
    m2 = theano.shared(np.random.randn(8,32).astype(np.float32))

    loss = (X - T.dot(T.dot(X,m1),m2)).norm(L=2)
    params = [m1,m2]
    updates = OrderedDict()
    lr0 = .001
    lr1 = theano.shared(np.float32(.001))
    print(lr1.get_value().dtype)
    grads = T.grad(loss, params)
    for g,p in zip(grads, params):
        updates[p] = p - lr1*g
    train = theano.function(inputs=[X], outputs=loss, updates=updates)
    for i in range(100):
        print(train(x))

test()
```

**Comment by nouiz** [comment]
I'm able to reproduce it. It is a harmless float64. Sadly this happen from time to time on the new back-end, as it now support float64 and sometimes, scalar end up in float64 for some reasons.

You can ignore it (so use warn_float64=warn instead of raise for now).

I'll try to fix that. thanks for the report.

**Comment by nouiz** [comment]
@lamblin @abergeron what do you think of the option at the end?
While investigating this I found that we moved too much elemwise of scalar to the GPU. This is fixed by this PR:

https://github.com/Theano/Theano/pull/5774

That do not fix the original problem. With this PR, the problem is that the elemwise fusion opt will fuse a cast to float64 introduced by GpuGemm make_node.

It is getting hard to make all those extra float64 not trigger warn_float64. I see 3 options:

- Continue as now and patch manually each cases.
- Disable warn_float64 during compilation by default by moving that to a new flag compile_warn_float64
- Stop using float64 for some ops in the graph when not needed.

I would go with option 2.

**Comment by abergeron** [comment]
I would go for option 2 or option 4 which is: remove the warn_float64 flag completely since the problem it tries to address is no longer present in the new backend.

**Comment by nouiz** [comment]
I got discussion with @abergeron live and we found that we didn't need the float64 in the graph. It was there to make the code faster to implement. So we dicided to go with option 3.

**Comment by nouiz** [comment]
This PR should fix this: https://github.com/Theano/Theano/pull/5774
